rtb/norme_di_progetto_ver0.5.0.typ

```diff
@@ -175,6 +184,92 @@ Nella fase di implementazione, il fornitore deve stabilire o selezionare un mode
+
+Esempio: Dopo un pagamento andato a buon fine, il sistema aggiorna lo stato dell'ordine a 'confermato' e invia una ricevuta via email.
+
+===== Trigger
+Il _trigger_ rappresenta l'evento o l'azione che scatena il caso d'uso. Questo può essere un'azione compiuta da un attore, come un _clic_ su un pulsante, oppure un evento esterno come il ricevimento di un messaggio da un altro sistema.
+
+Esempio: L'utente clicca sul pulsante 'Effettua pagamento'.
+
+===== Scenario principale
+Lo scenario principale descrive la sequenza di azioni _standard_ che l'attore ed il sistema eseguono per raggiungere l'obiettivo del caso d'uso. Include tutti i passi principali in ordine sequenziale.
+
+Esempio:
+1. L'utente inserisce i dati di accesso.
+2. Il sistema verifica le credenziali.
+3. L'utente visualizza la propria _dashboard_.
+
+===== Relazioni
+Nei casi d'uso possiamo avere i seguenti tipi di relazione tra attore e caso d'uso.
+
+====== Associazione 
+Congiunge semplicemente gli attori con i casi d'uso a cui prendono parte. Un attore può essere associato a qualsiasi numero di casi d'uso e viceversa. L'associzione implica uno scambio di informazioni tra attore e caso d'uso associato. Viene indicata nel diagramma con una linea nera.
+
+====== Generalizzazione
+Riguarda sia gli attori che i casi d'uso e rappresenta una relazione tra una classe, un caso d'uso o un attore più generale (superclasse) ed uno più specifico (sottoclasse). Questo tipo di relazione segue il principio dell'ereditarietà, in cui la sottoclasse eredita le caratteristiche dalla superclasse e può aggiungere o modificare specifici dettagli. La generalizzazione si rappresenta con una linea che collega le due entità (classi, attori o casi d'uso) e termina con un triangolo vuoto sulla punta, che indica la direzione verso la classe o entità più generale.
+
+Esempi: 
+- Un attore generico, come Utente, può essere specializzato in Cliente e Amministratore.
+- Un caso d'uso generico, come Gestire account, può essere specializzato in Creare account e Modificare account.
+
+
+====== Inclusione
+Le inclusioni rappresentano funzionalità o comportamenti comuni che possono essere riutilizzati in più casi d'uso per evitare duplicazioni. Viene indicata nel diagramma come una freccia tratteggiata con indicato "_include_". La funzione alla base della freccia include completamente la funzione alla punta della freccia.
+
+Esempio: Un caso d'uso Autenticazione utente può essere incluso in casi d'uso come Acquista prodotto o Gestione profilo.
+
+====== Estensione
+Le estensioni definiscono variazioni opzionali o eccezioni al comportamento principale del caso d'uso. Questi flussi si attivano solo se si verificano determinate condizioni. Viene indicata nel diagramma come una freccia tratteggiata con indicato "_extend_". La funzione alla base della freccia può essere impiegata nel contesto della funzione alla punta della freccia. 
```

Comment:
<\<extend>> (in typst mi sembra che già i doppi apici vengano convertiti in doppie triangolari